patches/tModLoader/Terraria/Program.TML.cs

```diff
@@ -87,24 +87,14 @@ public static partial class Program
 		}
 
 		private static void Port143FilesFromStable(string savePath, bool isCloud) {
-			// Copy all current stable player files to 1.4.3-legacy during transition period
+			// Copy all current stable player files to 1.4.3-legacy during transition period. Skip ModSources & Workshop shared folders
 
 			string newFolderPath = Path.Combine(savePath, ReleaseFolder);
 			string oldFolderPath = Path.Combine(savePath, StableFolder);
 
 			if (!Directory.Exists(newFolderPath) && Directory.Exists(oldFolderPath)) {
-				
-				Utilities.FileUtilities.CopyFolder(oldFolderPath, newFolderPath, isCloud);
-
-				// Windows doesn't like deleting things right after a copy call. It tends to lag behind.
-				// Especially if OneDrive is involved. This is not a good solution and will fail as is
-				string modSources = Path.Combine(newFolderPath, "ModSources");
-				if (Directory.Exists(modSources))
-					Directory.Delete(modSources, true);
-
-				string workshopTemp = Path.Combine(newFolderPath, "Workshop");
-				if (Directory.Exists(workshopTemp))
-					Directory.Delete(workshopTemp, true);
+				Utilities.FileUtilities.CopyFolder(oldFolderPath, newFolderPath, isCloud,
+					excludeFilter: new System.Text.RegularExpressions.Regex("(Workshop)|(ModSources)"));
```

Comment:
I'm concerned about an exclude filter which doesn't have locators. `Directory.GetDirectories` returns full paths, and `IsMatch` with this regex will find the specified string at any place in the path. Something like `WorkshopInfo.json` could end up getting excluded.

Reply:
Make it more strict - Looks for /Workshop/ or /ModSources/.
This will ensure it only excludes items deriving those intermediate folders